pkg/piecestore/psserver/retrieve.go

```diff
@@ -168,18 +168,18 @@ func (s *Server) retrieveData(ctx context.Context, stream pb.PieceStoreRoutes_Re
 
 		used += nextMessageSize
 		n, err := io.CopyN(writer, storeFile, nextMessageSize)
+		if err != nil && err != io.EOF {
```

Comment:
Doesn't this now end up in infinite loop when file returns io.EOF and we haven't reached the length?

Reply:
I think on error we should still increase allocationTracking. I suspect it will cause problems otherwise.

Reply:
Hmm... I don't know.

I also read in the `io.CopyN` that 

> On return, written == n if and only if err == nil

So, it's seems I am fixing the wrong thing.

Reply:
Move the EOF check here to  where we check the err from `s.retrieveData`

Reply:
I pushed another commit that makes sure we pass the correct number of bytes to CopyN. This way the EOF check is not necessary.